src/components/Feed/Feed.tsx

```diff
@@ -0,0 +1,150 @@
+import ActionText from 'components/core/ActionText/ActionText';
+import breakpoints from 'components/core/breakpoints';
+import colors from 'components/core/colors';
+import { fadeIn } from 'components/core/keyframes';
+import Spacer from 'components/core/Spacer/Spacer';
+import transitions from 'components/core/transitions';
+import NavbarGLink from 'components/NavbarGLink';
+import { useGlobalLayoutActions } from 'contexts/globalLayout/GlobalLayoutContext';
+import { useCallback, useEffect, useState } from 'react';
+import { graphql, useLazyLoadQuery } from 'react-relay';
+import styled from 'styled-components';
+import { FeedViewerQuery } from '__generated__/FeedViewerQuery.graphql';
+import GlobalFeed from './GlobalFeed';
+import ViewerFeed from './ViewerFeed';
+
+export const FOLLOWING = Symbol('FOLLOWING');
+export const WORLDWIDE = Symbol('WORLDWIDE');
+export type FeedMode = typeof FOLLOWING | typeof WORLDWIDE;
+
+type ControlProps = {
+  setFeedMode: (mode: FeedMode) => void;
+  initialFeedMode: FeedMode;
+};
+
+function FeedNavbarControl({ setFeedMode, initialFeedMode }: ControlProps) {
+  // Internally mirror the feed mode state so that we can avoid adding it to the dep array, and avoid re-adding these controls to the navbar when the user switches from Following to Worldwide feed
+  const [feedModeCopy, setFeedModeCopy] = useState<FeedMode>(initialFeedMode);
+
+  const handleFollowingModeClick = useCallback(() => {
+    setFeedMode(FOLLOWING);
+    setFeedModeCopy(FOLLOWING);
+  }, [setFeedMode]);
+
+  const handleWorldwideModeClick = useCallback(() => {
+    setFeedMode(WORLDWIDE);
+    setFeedModeCopy(WORLDWIDE);
+  }, [setFeedMode]);
```

Comment:
hmm a bit confused by the state duplication.

could do something like this
```js
function feedNavbarControl({ feedMode, setFeedMode }) { // get state/setter passed in from parent
  ...
  <StyledNavControlText
    onClick={() => setFeedMode(FOLLOWING)}
    focused={feedMode === FOLLOWING}
  />
  ...
}
```

then we wouldn't need local state in this file. also should be fine to not use `useCallback` since the Text element is super basic

Reply:
unfortunately because this component gets added to the navbar via `setCustomNavCenterContent()` it doesn't seem to get the updated `feedMode` value if it changes

Reply:
if you know of a workaround that would be awesome

Reply:
might have to just re-add this component to the navbar whenever the user switches the mode..